client/packages/invoices/src/Returns/InboundDetailView/Toolbar.tsx

```diff
@@ -19,14 +20,20 @@ import { InboundReturnFragment, useReturns } from '../api';
 export const Toolbar: FC = () => {
   const t = useTranslation('distribution');
   const onDelete = useReturns.lines.deleteSelectedInboundLines();
+  const { debouncedMutateAsync } = useReturns.document.updateInboundReturn();
+  const isDisabled = useReturns.utils.inboundIsDisabled();
+
   const { data } = useReturns.document.inboundReturn();
-  const { otherPartyName } = data ?? {};
-  const { isGrouped, toggleIsGrouped } = useIsGrouped('inboundReturn');
-  //   const [theirReferenceBuffer, setTheirReferenceBuffer] =
-  //     useBufferState(theirReference);
-  //   const { mutateAsync: updateName } = useOutbound.document.updateName();
+  const { otherPartyName, theirReference, id } = data ?? {};
 
-  //   const isDisabled = useOutbound.utils.isDisabled();
+  const update = (data: Partial<InboundReturnFragment>) => {
+    if (!id) return;
+    debouncedMutateAsync({ id, ...data });
+  };
+
+  const { isGrouped, toggleIsGrouped } = useIsGrouped('inboundReturn');
+  const [theirReferenceBuffer, setTheirReferenceBuffer] =
+    useBufferState(theirReference);
```

Comment:
I am not a fan of buffer state. Btw I get the same issue as the comment (due to useBufferState I think).

What we really need is a useDebunceState (a hook to call update method and to hold the 'value'):
```typescript
const [theirReferenceBuffer, setTheirReferenceBuffer] = useDebounceState(theirReference, (theirReference) => { patch({theirReference})})

```